filament/backend/src/opengl/PlatformGLX.cpp

```diff
@@ -110,14 +121,62 @@ Driver* PlatformGLX::createDriver(void* const sharedGLContext) noexcept {
         exit(1);
     }
 
-    // Create a context
-    static int attribs[] = { GLX_DOUBLEBUFFER, True, None };
+    if (nullptr != sharedGLContext) {
+      
+        int r = -1;
+        int used_fb_id = -1;
+        GLXContext shared_ctx = (GLXContext)((void*)sharedGLContext);
+      
+        r = g_glx.queryContext(mGLXDisplay, shared_ctx, GLX_FBCONFIG_ID, &used_fb_id);
+        if (0 != r) {
+            printf("Error: failed to get GLX_FBCONFIG_ID from shared GL context.");
+            return nullptr;
+        }
+    
+        int num_configs = 0;
+        GLXFBConfig* fb_configs = g_glx.getFbConfigs(mGLXDisplay, 0, &num_configs);
+
+        if (nullptr == fb_configs) {
+            printf("Failed to get the available GLXFBConfigs.\n");
+            return nullptr;
+        }
+
+        int fb_id = 0;
+        int fb_index = -1;
+    
+        for (int i = 0; i < num_configs; ++i) {
+        
+            r = g_glx.getFbConfigAttrib(mGLXDisplay, fb_configs[i], GLX_FBCONFIG_ID, &fb_id);
+            if (0 != r) {
+                printf("Error: failed to get GLX_FBCONFIG_ID for entry %d.\n", i);
+                continue;
+            }
+        
+            if (fb_id == used_fb_id) {
+                fb_index = i;
+                break;
+            }
+        }
+
+        if (fb_index < 0) {
+            printf("Error: failed to find an `GLXFBConfig` with the requested ID. (exiting).\n");
+            return nullptr;
+        }
+      
+        mGLXConfig = fb_configs + fb_index;
+    }
+
+    if (nullptr == sharedGLContext) {
+
+      // Create a context
+      static int attribs[] = { GLX_DOUBLEBUFFER, True, None };
 
-    int config_count = 0;
-    mGLXConfig = g_glx.chooseFbConfig(mGLXDisplay, DefaultScreen(mGLXDisplay),
-            attribs, &config_count);
-    if (mGLXConfig == nullptr || config_count == 0) {
+      int config_count = 0;
+      mGLXConfig = g_glx.chooseFbConfig(mGLXDisplay, DefaultScreen(mGLXDisplay),
+                                        attribs, &config_count);
```

Comment:
Continuation indents are 8 spaces in our code base, we don't align parameters in this way

Reply:
Done.